### ISSUE TITLE
help: How do I center the header?

### ISSUE THREAD

**Issue by hyawara** [MeanderingProgrammer/render-markdown.nvim#179]
### Neovim version (nvim -v)

0.10.0

### Neovim distribution

N/A

### Description

I want to center the title, like this:

![image](https://github.com/user-attachments/assets/750ef1c7-ba35-4b2f-b0ec-26e2c49c4b8a)

**Comment by MeanderingProgrammer** [comment]
There is no way to do this. You can specify a `left_pad` value for headings but that will apply a constant padding to all heading levels and naturally will not change when the window resizes.

I'm not really aware of a markdown renderer that supports this kind of thing and in general I don't think I want to support centering content mostly because it would need to interoperate with all the existing options. I gave implementing it a shot and it ended up messy with a lot of gaps.

If I made `left_pad` accept a function that takes a `window_id` and the current `line` of text would that be enough?

I imagine the implementation in your config would look something like:

```lua
require('render-markdown').setup({
  heading = {
    left_pad = function(win, line)
      local window_width = return vim.api.nvim_win_get_width(win)
      local line_width = vim.fn.strdisplaywidth(line)
      return math.floor((window_width - line_width) / 2)
    end,
  },
})
```

Basically any weird offsets and things specific to your config just end being something you solve rather than me needing to handle more edge cases.

**Comment by MeanderingProgrammer** [comment]
I actually ended up implementing this internally here: https://github.com/MeanderingProgrammer/render-markdown.nvim/commit/0986638b381a4b01eb108bb946f3a67a9eb3d0ec

It applies to both headings and code blocks, since they're alignment configs are the same.

By providing floating point values between 0 & 1 to `left_pad`, `right_pad` and `left_margin` you can get different effects.

Here are a few configs to try:

```lua
require('render-markdown').setup({
  heading = {
    left_pad = 0.5,
  },
})
```

```lua
require('render-markdown').setup({
  heading = {
    left_pad = 0.5, 
    position = 'inline',
  },
})
```

```lua
require('render-markdown').setup({
  heading = {
    left_margin = 0.5,
    position = 'inline',
    width = 'block',
  },
})
```

```lua
require('render-markdown').setup({
  heading = {
    left_margin = 0.5,
    position = 'inline',
    width = 'block',
    left_pad = 0.2,
    right_pad = 0.2,
  },
})
```

**Comment by tristone13th** [comment]
Just want to know can we have a specific type of header (e.g., header 1) centered without touching other headers' style?

**Comment by MeanderingProgrammer** [comment]
You can provide a list of values to the various properties, the last value gets used for all remaining headers.

So to center just h1:

```lua
require('render-markdown').setup({
  heading = {
    left_pad = { 0.5, 0 },
    -- This is equivalent to:
    -- left_pad = { 0.5, 0, 0, 0, 0, 0 },
  },
})
```

To center h2 & h4 only:

```lua
require('render-markdown').setup({
  heading = {
    left_pad = { 0, 0.5, 0, 0.5, 0 },
  },
})
```
